packages/gatsby/src/redux/actions/public.js

```diff
@@ -172,97 +174,110 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
             `
       if (process.env.NODE_ENV === `test`) {
         return error
-        // Only error if the context version is different than the page
-        // version.  People in v1 often thought that they needed to also pass
-        // the path to context for it to be available in GraphQL
-      } else if (invalidFields.some(f => page.context[f] !== page[f])) {
-        report.panic(error)
       } else {
-        if (!hasWarnedForPageComponentInvalidContext.has(page.component)) {
-          report.warn(error)
-          hasWarnedForPageComponentInvalidContext.add(page.component)
-        }
+        report.panic({
```

Comment:
Is deduping not necessary anymore?

(Like we have in `hasWarnedForPageComponentInvalidContext` in the before)

Reply:
I just changed it to error every time instead of warning.

I added this code during pre-v2 as there'd been a bad pattern that had developed with people creating pages during v1 and I wanted to warn people against it. So I made it a warning during development and error during builds. It's been long enough that I thought just simplifying things to always error would work.

Reply:
We'll probably want to not change this behaviour (warn versus error) as a hard fail would be a breaking change IMO

As a follow up to that, if we're warning, we'll likely want to keep the dedupe logic